cpmpy/expressions/variables.py

```diff
@@ -263,11 +266,23 @@ def __init__(self, lb, ub, name):
         self.name = name
         self._value = None
 
+    def has_subexpr(self):
+        """Does it contains nested Expressions?
+           Is of importance when deciding whether transformation/decomposition is needed.
+        """
+        return False
+
     def is_bool(self):
         """ is it a Boolean (return type) Operator?
         """
         return False
 
+    def is_leaf(self):
+        """ Is it the leaf of an expression tree?
+            This is only the case for decision variables (and constants).
+        """
+        return True
```

Comment:
we can build an cpm_array of expressions... I don't think it is generally true to be a leaf?

Reply:
this is another part of the code, not ndvararray